src/Storages/StoragePostgreSQL.cpp

```diff
@@ -456,15 +458,20 @@ SinkToStoragePtr StoragePostgreSQL::write(
     return std::make_shared<PostgreSQLSink>(metadata_snapshot, pool->get(), remote_table_name, remote_table_schema, on_conflict);
 }
 
-StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, bool require_table)
+StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, PostgreSQLSettings& storage_settings, bool require_table)
 {
     StoragePostgreSQL::Configuration configuration;
+
+    ValidateKeysMultiset<ExternalDatabaseEqualKeysSet> optional_arguments = {"replace_query", "on_duplicate_clause", "addresses_expr", "host", "hostname", "port"};
+    auto postgresql_settings = storage_settings.all();
+    for (const auto & setting : postgresql_settings)
+        optional_arguments.insert(setting.getName());
     ValidateKeysMultiset<ExternalDatabaseEqualKeysSet> required_arguments = {"user", "username", "password", "database", "db"};
     if (require_table)
         required_arguments.insert("table");
 
     validateNamedCollection<ValidateKeysMultiset<ExternalDatabaseEqualKeysSet>>(
-        named_collection, required_arguments, {"schema", "on_conflict", "addresses_expr", "host", "hostname", "port", "use_table_cache"});
```

Comment:
This change broke integration tests. This list should be moved in line 465 instead of the list you have there (looks like you put there MySQL engine optional arguments, but they are different for PostgreSQL arguments)